src/VisualStudio/Core/Def/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -149,13 +152,27 @@ internal sealed class SourceGeneratedFileManager : IOpenTextBufferEventListener
 
             identity = default;
 
+            // Find GUID in "temporary directory/GUID/path/to/file.cs".
             if (!filePath.StartsWith(_temporaryDirectory))
             {
                 return false;
             }
 
-            var fileInfo = new FileInfo(filePath);
-            return Guid.TryParse(fileInfo.Directory.Name, out var guid) &&
+            var slice = filePath.AsSpan()[_temporaryDirectory.Length..];
+            if (slice.IsEmpty || !PathUtilities.IsDirectorySeparator(slice[0]))
```

Comment:
Should we just ensure that _temporaryDirectory always has a trailing slash? Then it'd remove this need to check the directory separator and also the extra slice operation there too.